### PR TITLE
Convert auto-bucket-datetimes middleware to MLv2

### LINKED ISSUE #39834: [MLv2 × QP] Migrate pre-processing middleware `metabase.query-processor.middleware.auto-bucket-datetimes`

### PR DESCRIPTION
**Pull request by camsaul**
Resolves #39834

Internal refactor. The main reason for this refactor is to make some other bugfixes I'm working on easier to do -- code is a lot simpler when using MLv2 internally.

Bonus stuff:

- I also added some new Kondo rules to enforce usage of the `lib.schema` or `legacy-mbql.schema` schemas in `metabase.lib` and the QP, as opposed to the ones used in the REST API/model stuff in `metabase.util.malli.schema` -- these versions are preferrable because they have the extra documentation added and keep `metabase.lib`'s dependencies on other parts of the codebase to a minimum.
- I enabled the prefer `update-keys` and `update-vals` over the Medley versions in the QP.
- I removed `qp.store/bulk-metadata` in favor of `lib.metadata/bulk-metadata-or-throw`, moving us one step closer to removing the QP store entirely (once all QP middleware uses MLv2 we can use the query itself for metadata)

### CODE REVIEW

File: modules/drivers/googleanalytics/test/metabase/driver/googleanalytics_test.clj [hunk 1]
```diff
@@ -243,12 +243,11 @@
...
                                                (t/local-time 0)
                                                (t/zone-id system-timezone-id)))
-                                  (t/zone-id system-timezone-id))
+                                   (t/zone-id system-timezone-id))
         (is (= expected-ga-query
                (do-with-some-fields
-                (fn [{:keys [db event-action-field event-label-field date-field], :as objects}]
+                (fn [{:keys [db], :as objects}]
                   (qp.store/with-metadata-provider (u/the-id db)
-                    (qp.store/bulk-metadata :metadata/column (map u/the-id [event-action-field event-label-field date-field]))
```
Comment (metamben) [inline modules/drivers/googleanalytics/test/metabase/driver/googleanalytics_test.clj:251]: Why is this removed instead of inlined?

**Review by metamben — APPROVED** [review]
LGTM

File: modules/drivers/googleanalytics/test/metabase/driver/googleanalytics_test.clj (hunk 1 above)
Comment (camsaul) [reply modules/drivers/googleanalytics/test/metabase/driver/googleanalytics_test.clj:251]: it was only done for side effects, the old QP store code (before we switched the QP to MLv2 metadata providers) would error if you tried to access something that wasn't already fetched. The new MetadataProviders just fetch it instead of complaining. This line is left over from the PR that switched everything to MetadataProviders, we're not getting any benefit from it except maybe saving one app DB call in a test
